Sources/NIO/EventLoopFuture.swift

```diff
@@ -793,7 +793,29 @@ extension EventLoopFuture {
         p0.succeed(result: ())
         return body
     }
+}
 
+extension EventLoopFuture {
+    /// Returns an `EventLoopFuture` that fires when this future completes, but executes its callbacks on the
+    /// target event loop instead of the original one.
+    ///
+    /// It is common to want to "hop" event loops when you arrange some work: for example, you're closing one channel
+    /// from another, and want to hop back when the close completes. This method lets you spell that requirement
+    /// succinctly. It also contains an optimisation for the case when the loop you're hopping *from* is the same as
+    /// the one you're hopping *to*, allowing you to avoid doing allocations in that case.
+    ///
+    /// - parameters:
+    ///     - target: The `EventLoop` that the returned `EventLoopFuture` will run on.
+    /// - returns: An `EventLoopFuture` whose callbacks run on `target` instead of the original loop.
+    func on(eventLoop target: EventLoop) -> EventLoopFuture<T> {
```

Comment:
I'm not sure about this name. Maybe `hopTo(eventLoop:)`?

Reply:
Like this PR! What about just `hop(eventLoop: EventLoop)`. So `future.hop(eventLoop: otherEL)`.

Reply:
I think I would either call it `hopTo(eventLoop:)` or just use `hopOn(eventLoop:)`.

Reply:
@weissi Any objection to `hopTo`?

Reply:
Nope, sounds good to me